### PR TITLE
feat: Add arm64 fp16 rotate support

### PR DESCRIPTION
**Pull request by Xinyu302**
Add arm64 fp16 rotate and test. OSPP project implementation.

### CODE REVIEW

**Comment by chenqy4933** [comment]
How about the performance of arm64 optimized kernel vs before?

**Comment by chenqy4933** [comment]
LGTM

**Comment by Xinyu302** [comment]
> How about the performance of arm64 optimized kernel vs before?

There is no arm64 fp16 rotate kernel before. I only add a test in compiler/test/kernel/opr/arm/cv.cpp and pass the correctness test.

I would appreciate it if you can tell me how can I get a kernel's performance in megcc?

**Comment by Xinyu302** [comment]
It seems that CI/lint failed, I think it's not about me.

**Comment by chenqy4933** [comment]
@Xinyu302 there are some conflicts, please resolve them and re-run the ci

**Comment by Xinyu302** [comment]
> @Xinyu302 there are some conflicts, please resolve them and re-run the ci

OK.

**Comment by chenqy4933** [comment]
@Li-Ming-xin

File: compiler/lib/KernelGen/Arm/Arm64/Rotate.cpp [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+    return GetCVKernelSymbol(context) +
+           "(const TinyMat* src, const TinyMat* dst, bool clockwise)";
+}
+
+std::string RotateKernel::GetCVKernelBody(TContext* context) const {
+    auto kernel_sig = GetCVKernelSignature(context);
+    std::string body_temp = R"(
+        #include <arm_neon.h>
+        #include <string.h>
+        #include <stdio.h>
```
Comment (Li-Ming-xin) [inline compiler/lib/KernelGen/Arm/Arm64/Rotate.cpp:36]: `stdio.h`是之前用于调试的吗？应该需要删掉。 @Xinyu302

**Comment by Li-Ming-xin** [comment]
需要在`runtime/include/tinycv_c.h`添加rotate cv算子的函数声明，否则用户实际使用时会导致编译失败。

**Comment by Li-Ming-xin** [comment]
https://github.com/MegEngine/MegCC/blob/eabc1d908eed3796cba6b92d8a7e36ddd2e990c6/compiler/tools/mgb-to-tinynn/mgb-to-tinynn.cpp#L73C5-L73C5 这里应该缺少一些代码，可自行编译mgb-to-tinynn测试一下。 @Xinyu302 
ref: 
https://github.com/MegEngine/MegCC/blob/main/compiler/README.md#building 
https://github.com/MegEngine/MegCC/blob/main/doc/how-to-use-chinese.md#编写-json-文件

**Comment by Xinyu302** [comment]
> 需要在`runtime/include/tinycv_c.h`添加rotate cv算子的函数声明，否则用户实际使用时会导致编译失败。

已经完成了添加

**Comment by Xinyu302** [comment]
> https://github.com/MegEngine/MegCC/blob/eabc1d908eed3796cba6b92d8a7e36ddd2e990c6/compiler/tools/mgb-to-tinynn/mgb-to-tinynn.cpp#L73C5-L73C5 这里应该缺少一些代码，可自行编译mgb-to-tinynn测试一下。 @Xinyu302 ref: https://github.com/MegEngine/MegCC/blob/main/compiler/README.md#building [https://github.com/MegEngine/MegCC/blob/main/doc/how-to-use-chinese.md#编写-json-文件](https://github.com/MegEngine/MegCC/blob/main/doc/how-to-use-chinese.md#%E7%BC%96%E5%86%99-json-%E6%96%87%E4%BB%B6)

改写yolox_cv.json 为：
```json
{
    "dump_dir":"./kernel_yolox_s_arm/",
    "models":[
        {
            "model_name":"yolox_s",
            "model_path":"./yolox_s.mge",
            "input_shape_str":"data=(1,3,640,640)",
            "enable_nchw44":true
        }
    ],
    "cv":{
        "transpose":["ui8"],
        "roicopy":["ui8"],
        "rotate":["f16"],
        "flip":["ui8"],
        "resize_linear":["ui8"],
        "warp_affine_replicate_linear":["ui8"],
        "rgb2bgr":["ui8"],
        "yuv2bgr_nv21":["ui8"],
        "rgb2yuv":["ui8"]
    }
}
```

使用命令
```bash
mgb-to-tinynn --json=yolox_cv.json --arm64 --dump yolox_gen
```
进行测试，在yolox_gen文件夹中生成了tinycv_rotate_f16.c

**Comment by Li-Ming-xin** [comment]
> 使用命令
> 
> ```shell
> mgb-to-tinynn --json=yolox_cv.json --arm64 --dump yolox_gen
> ```
> 
> 进行测试，在yolox_gen文件夹中生成了tinycv_rotate_f16.c

ok，我去确认了一下，之前确实已经有rotate的相关代码了。

**Comment by Li-Ming-xin** [comment]
LGTM

**Review by Li-Ming-xin — APPROVED** [review]
LGTM

**Comment by Xinyu302** [comment]
根据OSPP的要求，提PR的邮箱需要和OSPP注册邮箱一致，本PR关闭并重新整理提交新的PR
@Li-Ming-xin

**Pull request closed without merging**